src/Datagrid/Pager.php

```diff
@@ -222,7 +226,19 @@ public function getLinks($nbLinks = null)
      */
     public function haveToPaginate()
     {
-        return $this->getMaxPerPage() && $this->getNbResults() > $this->getMaxPerPage();
+        // NEXT_MAJOR: remove the existence check and the else part
+        if (method_exists($this, 'countResults')) {
+            $countResults = (int) $this->countResults();
```

Comment:
```suggestion
            $countResults = $this->countResults();
```

This is not needed, isn't it?

Reply:
See https://github.com/sonata-project/SonataAdminBundle/pull/6732#discussion_r551025466